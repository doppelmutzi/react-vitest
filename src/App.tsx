import { useEffect, useState } from 'react'
import reactLogo from './assets/react.svg'
import vitestLogo from '/vitest.svg'
import './App.css'
import { fetchQuote, Quote } from './quote.service'

function App() {
  const [count, setCount] = useState(0)


  const [quote, setQuote] = useState<Quote>();
  useEffect(() => {
    const getQuote = async () => {
      const quote = await fetchQuote();
      setQuote(quote);
    };
    getQuote();
  }, []);
  

  return (
    <>
      <div>
        <a href="https://vitest.dev/" target="_blank">
          <img src={vitestLogo} className="logo" alt="Vitest logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vitest + React ü§ù</h1>
      <section>
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <h2>Quote of the Day</h2>
        <p>{quote ? quote.quote : 'Loading...'}</p>
      </section>
    </>
  )
}

export default App
