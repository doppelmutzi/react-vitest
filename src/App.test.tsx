import { describe, expect, test } from "vitest";
import App from "../src/App";
import { fireEvent, render, screen } from "@testing-library/react";

describe("App", () => {
  test("renders headline", () => {
    render(<App />);
    const button = screen.getByText("count is 0");
    fireEvent.click(button);
    expect(screen.getByText("count is 1")).toBeDefined();
    fireEvent.click(button);
    expect(screen.getByText("count is 2")).toBeDefined();
  });

  test("matches snapshot", () => {
    const { container, getByText } = render(<App />);
    expect(container).toMatchSnapshot();
    const button = getByText("count is 0");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});
