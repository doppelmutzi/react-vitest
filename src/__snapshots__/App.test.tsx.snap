// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches snapshot 1`] = `
<div>
  <div>
    <a
      href="https://vitest.dev/"
      target="_blank"
    >
      <img
        alt="Vitest logo"
        class="logo"
        src="/vitest.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vitest + React ü§ù
  </h1>
  <section>
    <button>
      count is 
      0
    </button>
    <h2>
      Quote of the Day
    </h2>
    <p>
      Loading...
    </p>
  </section>
</div>
`;

exports[`App > matches snapshot 2`] = `
<div>
  <div>
    <a
      href="https://vitest.dev/"
      target="_blank"
    >
      <img
        alt="Vitest logo"
        class="logo"
        src="/vitest.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vitest + React ü§ù
  </h1>
  <section>
    <button>
      count is 
      1
    </button>
    <h2>
      Quote of the Day
    </h2>
    <p>
      Loading...
    </p>
  </section>
</div>
`;
